lib/ecto/query/planner.ex

```diff
@@ -1732,7 +1732,13 @@ defmodule Ecto.Query.Planner do
         type
 
       Map.has_key?(schema.__struct__(), field) ->
-        error! query, expr, "field `#{field}` in `#{kind}` is a virtual field in schema #{inspect schema}"
+        case schema.__schema__(:association, field) do
+          %Ecto.Association.BelongsTo{owner_key: owner_key} ->
```

Comment:
What about other types of association? Maybe we could give a specific message for them?